src/repr/src/row/encoding.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A permanent storage encoding for rows.
+//!
+//! To minimize cycles spent encoding/decoding, we use Row's internal storage
+//! format, with a prefix to allow for migrations if the internal format
+//! changes.
+//!
+//! ```none
+//! <version: 0u8><v0_encoding: &[u8]>
```

Comment:
We could save the version byte on each key and value if we versioned the row format at the "batch" level, but (1) that's certainly action at a distance (2) our batch format will probably in flux for longer than this and (3) not sure I'm convinced the saved overhead is worth it.